streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -198,8 +221,58 @@ public void shouldRestoreMultipleStores() throws Exception {
     }
 
     @Test
+    public void shouldRestoreAndNotifyMultipleStores() throws Exception {
+        final TopicPartition one = new TopicPartition("one", 0);
+        final TopicPartition two = new TopicPartition("two", 0);
+        final MockStateRestoreListener callbackOne = new MockStateRestoreListener();
+        final MockStateRestoreListener callbackTwo = new MockStateRestoreListener();
+        final CompositeRestoreListener restoreListener1 = new CompositeRestoreListener(callbackOne);
+        final CompositeRestoreListener restoreListener2 = new CompositeRestoreListener(callbackTwo);
+        setupConsumer(10, topicPartition);
+        setupConsumer(5, one);
+        setupConsumer(3, two);
+
+        changelogReader
+            .register(new StateRestorer(topicPartition, restoreListener, null, Long.MAX_VALUE, true, "storeName1"));
+        changelogReader.register(new StateRestorer(one, restoreListener1, null, Long.MAX_VALUE, true, "storeName2"));
+        changelogReader.register(new StateRestorer(two, restoreListener2, null, Long.MAX_VALUE, true, "storeName3"));
+
+        changelogReader.restore();
+
+        assertThat(callback.restored.size(), equalTo(10));
+        assertThat(callbackOne.restored.size(), equalTo(5));
+        assertThat(callbackTwo.restored.size(), equalTo(3));
+
+        assertThat(callback.storeNameCalledStates.get(RESTORE_START), equalTo("storeName1"));
```

Comment:
could we extract these 3 lines into a method, say `verifyCallbackStatesCalled` or something better! the same block of code is repeated 3 times in the test so would make it easier to grok